pkgs/build-support/setup-hooks/desktop-to-darwin-bundle.sh

```diff
@@ -150,7 +150,16 @@ convertDesktopFile() {
     local -r file=$1
     local -r sharePath=$(dirname "$(dirname "$file")")
     local -r name=$(getDesktopParam "${file}" "^Name")
-    local -r exec=$(getDesktopParam "${file}" "Exec")
+    local -r macOSExec=$(getDesktopParam "${file}" "X-macOS-Exec")
+    if [[ "$macOSExec" ]]; then
+      local -r exec="$macOSExec"
+    else
+      local -r execRaw=$(getDesktopParam "${file}" "Exec")
+      local -r exec="${execRaw// %[fFuUick]}"
+      if [[ "$exec" != "$execRaw" ]]; then
+        echo "desktopToDarwinBundle: Application bundles do not understand desktop entry field codes. Changed '$execRaw' to '$exec'."
```

Comment:
Should we consider `exit 1` here to stop the build?  My concern is that this message may get lost and then we would be executing something that isn't intended.

Reply:
Agreed.  To be fair, I hadn't looked up the spec on what these flags mean.  Having read through then now, I don't think they're applicable to MacOS.